### ISSUE TITLE
Training imagenet: loss does not decrease

### ISSUE THREAD

**Issue by to3i** [BVLC/caffe#401]
Hi, first of all thank you for sharing caffe, it really looks great! I run into an issue training the imagenet model. The training error is not decreasing even after 40k iterations. I am using the imagenet configuration provided in the examples folder. The model is trained with caffe from the dev branch (since master has not been merged with boost-eigen yet).

The train loss starts out around 7.1, then decreases close to 6.9 in the first 150 iterations, and then remains above that value for 40k iterations and likely beyond. I run the same configuration on the boost-eigen branch and had not trouble with the training. The error started decreasing after about 1k iterations and everything seems to be working fine.

Am I missing something? Can you reproduce this behavior? I am running caffe on Ubuntu 12.04 with Cuda 5.5 (GTX 770).

**Comment by jltmtz** [comment]
I have run the imagenet tutorial around 8 times now, and seen it stall 3/8 times. As mentioned by @sguada on #59, if there is no increment after 5000 iterations that's probably a good sign that the net won't converge, and you should restart it and hope for the best.

**Comment by lugiavn** [comment]
I'm running this, the loss stuck at 6.9 for 8k iteration before going down. But now it keeps going down so that's good.

**Comment by zizhaozhang** [comment]
I haven't test imagenet by myself yet. I re-implement a network in a paper that the loss does not go down. But it works when I use another deep learning framework (so my understanding of the paper and dataset preparation is correct). If I can get correct answer with mnist dataset. Could I say my cuda and caffe version is no problem?

**Comment by anurikadisha** [comment]
Hi @to3i 
I am also facing the same problem. Loss first decreases and then suddenly shoots up. How did you solve this problem?

**Comment by Dror370** [comment]
Hi all ,
I suggest you all to check the softmaxwithloss layer, see that you define the layer correctly,
In case you define Phase:train, it is not work properly, 
The layer definition should not include any phase ....
